MiniApp/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift

```diff
@@ -269,11 +275,29 @@ internal class MiniAppScriptMessageHandler: NSObject, WKScriptMessageHandler {
 class LocationManager: NSObject {
     let manager: CLLocationManager
     let location: CLLocation?
+    var locationListener: ((Result<CLLocation?, Error>) -> Void)?
 
     init(enableHighAccuracy: Bool) {
         manager = CLLocationManager()
         location = manager.location
         super.init()
         manager.desiredAccuracy = enableHighAccuracy ? kCLLocationAccuracyBest : kCLLocationAccuracyHundredMeters
+        manager.delegate = self
+    }
+
+    func updateLocation(result: @escaping (Result<CLLocation?, Error>) -> Void) {
+        if CLLocationManager.authorizationStatus() == .authorizedAlways || CLLocationManager.authorizationStatus() == .authorizedWhenInUse {
+            self.locationListener = result
+            manager.startUpdatingLocation()
+        } else {
+            result(.success(manager.location))
```

Comment:
I kept this as a success because unit test failures proved that this was the expected behavior. I'd prefer returning a failure, but we need to think about it before

Reply:
I think this is the wrong behavior so we should change this here. On Android, if the user tries to retrieve the location while the permission is denied, then the Android OS sends an error like: 
```
{code: 2, message: "application does not have sufficient geolocation permissions."}
```

It actually returns an error object - but I think we don't have that capability right now in the JS Bridge. So probably it's okay to just return the error message `application does not have sufficient geolocation permissions.`.